lib/OpenQA/Task/Job/Limit.pm

```diff
@@ -146,4 +151,106 @@ sub _limit_screenshots {
+        $margin_bytes = _check_remaining_disk_usage($job, $resultdir, $min_free_percentage);
+        return $margin_bytes >= 0;
+    };
+    $job->note(resultdir => $resultdir);
+    return $job->finish('Done, nothing to do') if $df_chk->();
+
+    # determine important builds (for each group)
+    # TODO: We should likely run the following code in a transaction like when computing the asset status to avoid reading
+    #       inconsistent data, e.g. if a new important build is scheduled during this cleanup we should not consider these
+    #       jobs unimportant just because the corresponding comment wasn't there at the time we started the cleanup.
+    # TODO: When we run this inside a transaction, would we still get a foreign key violation when deleting a screenshot
+    #       when a new screenshot link has been created in the meantime? Likely only at the time the transaction is commited
+    #       which would fail the entire transaction. Also not good.
+    # TODO: So maybe create a list of jobs which can be considered for sure upfront (inside a transaction) and do the actual
+    #       deletions later?
+    my $schema = $app->schema;
+    my $job_groups = $schema->resultset('JobGroups');
+    #my %important_builds_by_job_group_id = map { $_->id => $_->important_builds } $job_groups->all;
+    my %important_builds_hash;
+    for my $job_group ($job_groups->all) {
+        $important_builds_hash{$_} for @{$job_group->important_builds};
+    }
+    my @important_builds = keys %important_builds_hash;
+
+    # TODO: The subsequent cleanup simply deletes stuff from old jobs first. It does not take the retention periods
+    #       configured on job group level in account anymore. Is that good enough?
+    # TODO: I hope this doesn't explode. We're considering possibly lots of jobs at once here. Maybe we need to select
+    #       a range here.
+
+    log_debug "Deleting videos from non-important jobs startinng from oldest job (balance is $margin_bytes)";
+    my $jobs = $schema->resultset('Jobs');
+    my %jobs_search = (logs_present => 1, BUILD => {-not_in => \@important_builds});
+    my %jobs_params = (order_by => {-desc => 'id'});
+    my $relevant_jobs = $jobs->search({logs_present => 1, BUILD => {-not_in => \@important_builds}}, \%jobs_params);
+    while (my $job = $relevant_jobs->next) {
+        log_debug 'Deleting video of job ' . $job->id;
+        return $job->finish('Done after deleting videos from non-important jobs') if $margin_bytes += $job->delete_video;
+    }
+
+    log_debug "Deleting results from non-important jobs startinng from oldest job (balance is $margin_bytes)";
+    # TODO: Do we want to delete the whole job or only logs or all results but keep the DB entry? The phrasing
+    #       in the progress ticket sounds like the last option. However, we currently don't have this "grade of
+    #       cleanup" implemented so that would be something new.
+    # TODO: Implement delete_results.
```

Comment:
would you recommend to implement the "delete only results but keep job in db" first?

Reply:
I plan to make it an extra commit similar to `Add function to determine screenshots exclusively used by a job`. Same counts for the video deletion. (I split things up as soon as I have them tested.)

Reply:
Alright, sounds good. I suggest then to merge these features step by step if possible as likely something can break badly and we will have a hard time realizing … and then there is christmas vacation … ;)